dev/com.ibm.ws.security/src/com/ibm/ws/security/context/SubjectRegistryManager.java

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.security.context;
+
+import com.ibm.websphere.ras.annotation.Trivial;
+import com.ibm.ws.security.context.internal.SubjectRegistryThreadContext;
+
+/**
+ * The SubjectRegistryManager sets and gets caller/invocation subject information
+ * off the thread and provides the ability to clear the subject registry info off the thread.
+ * See {@link SubjectRegistryThreadContext} for more information.
+ */
+public class SubjectRegistryManager {
+
+    private static ThreadLocal<SubjectRegistryThreadContext> threadLocal = new SecurityThreadLocal();
+
+    /**
+     * Gets the subject registry thread context that is unique per thread.
+     * If/when a common thread storage framework is supplied, then this method
+     * implementation may need to be updated to take it into consideration.
+     *
+     * @return the subject registry thread context.
+     */
+    @Trivial
+    protected static SubjectRegistryThreadContext getSubjectRegistryThreadContext() {
+        ThreadLocal<SubjectRegistryThreadContext> currentThreadLocal = getThreadLocal();
+        SubjectRegistryThreadContext subjectRegistryThreadContext = currentThreadLocal.get();
+        if (subjectRegistryThreadContext == null) {
+            subjectRegistryThreadContext = new SubjectRegistryThreadContext();
+            currentThreadLocal.set(subjectRegistryThreadContext);
+        }
+        return subjectRegistryThreadContext;
+    }
+
+    /**
+     * Set whether or not the subject is from the SAF registry.
+     *
+     * @param isSAF True if the subject is from the SAF registry.
+     */
+    public static void setSubjectIsSAF(boolean isSAF) {
+        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
+        subjectRegistryThreadContext.setIsSAF(isSAF);
+    }
+
+    /**
+     * Start subject registry detection. This is only required if a SAF
+     * registry is configured.
+     *
+     * @param isSAFRegistryConfigured
+     */
+    public static void startSubjectRegistryDetectionOnZOS() {
+        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
```

Comment:
Probably do the `Platform.isZOS()` check here.

```suggestion
        if (!Platform.isZOS()) {
            return;
        }
        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
```